packages/tools/src/tools/FreehandROISculptorTool.ts

```diff
@@ -0,0 +1,685 @@
+    mouseLocation: undefined,
+  };
+  private _sculptData?: {
+    mousePoint: Types.Point3;
+    mouseCanvasPoint: Types.Point2;
+    points: Array<Types.Point3>;
+    toolSize: number;
+    maxSpacing: number;
+    element: HTMLDivElement;
+  };
+  toolSize: number;
+  isActive = false;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        maxToolSize: null,
+        minSpacing: 1,
+        referencedToolNames: [
+          'PlanarFreehandROI',
+          'PlanarFreehandContourSegmentationTool',
+        ],
+        referencedToolName: 'PlanarFreehandROI',
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+    this._endCallback = this._endCallback.bind(this);
+    this._dragCallback = this._dragCallback.bind(this);
+  }
+
+  preMouseDownCallback = (evt: EventTypes.InteractionEventType) => {
+    const eventData = evt.detail;
+    const element = eventData.element;
+
+    this._configureToolSize(evt);
+    this._selectFreehandTool(eventData);
+
+    if (this._commonData.selectedIndex === null) {
+      return;
+    }
+
+    this.isActive = true;
+
+    hideElementCursor(element);
+    this._activateSculpt(element);
+    return true;
+  };
+
+  mouseMoveCallback = (evt: EventTypes.InteractionEventType): void => {
+    if (this.mode === ToolModes.Active) {
+      this._configureToolSize(evt);
+      this._updateCursor(evt);
+    } else {
+      this._commonData.mouseLocation = undefined;
+    }
+  };
+
+  /**
+   * Updates cursor size
+   *
+   * @param evt - The event
+   */
+  _updateCursor(evt: EventTypes.InteractionEventType): void {
+    const eventData = evt.detail;
+    const element = eventData.element;
+    const config = this.configuration;
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine, viewport } = enabledElement;
+
+    this._commonData.viewportIdsToRender = [viewport.id];
+
+    const annotations =
+      this._filterInteractableFreehandRoiAnnotationsForElement(element);
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    this._commonData.mouseLocation = eventData.currentPoints.canvas;
+
+    if (this.isActive) {
+      annotations[this._commonData.selectedIndex].highlighted = true;
+    } else {
+      const canvasCoords = eventData.currentPoints.canvas;
+      const radius = this._distanceFromPoint(
+        viewport,
+        annotations[this._commonData.selectedIndex]?.data,
+        canvasCoords
+      );
+      if (radius > 0) {
+        this.toolSize = Math.min(config.maxToolSize, radius);
+      }
+    }
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      this._commonData.viewportIdsToRender
+    );
+  }
+
+  /**
+   * Returns interactable freehand ROI annotations
+   *
+   * @param element - The viewport element
+   */
+  _filterInteractableFreehandRoiAnnotationsForElement(element: HTMLDivElement) {
```

Comment:
Can we call this _filterSculptableAnnotationsForElement(...)
Just that we may well have more than just Freehand